x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/WithTargetNumberOfShards.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.ilm;
+
+/**
+ * The CheckTargetShardsCountStep requires for an action to provide the target number of shards. Any action
+ * implementing this interface can have the CheckTargetShardsCountStep step.
+ */
+public interface WithTargetNumberOfShards {
```

Comment:
Why do we need this interface? It seems we only have `ShrinkAction` implement it? Why not do the `instance of` check on `ShrinkAction`?

Reply:
Hm, I tried to generalize it because from the name of the step there is not association with the Shrink action, but we take the actual implementation of the step, I think it is actually pretty specific to the Shrink action. I will remove it.

Reply:
Removed